src/main/java/subway/domain/discountpolicy/BaseFarePolicy.java

```diff
@@ -0,0 +1,22 @@
+package subway.domain.discountpolicy;
+
+import subway.domain.Line;
+
+import java.util.List;
+
+public class BaseFarePolicy implements FarePolicy {
+
+    @Override
+    public int calculateFare(final List<Line> boardingLines, final int distance, final int age, final int fare) {
+        int total = fare + getMaxSurCharge(boardingLines);
+        total += Math.max(distance - 50, 0) / 8L * 100;
+        total += Math.max(distance - 10, 0) / 5L * 100;
+        return total;
+    }
+
+    private int getMaxSurCharge(final List<Line> boardingLines) {
+        return boardingLines.stream()
+                .mapToInt(Line::getSurcharge)
+                .max().getAsInt();
+    }
```

Comment:
해당 로직에서 max()의 최종반환은 OptionalInt 인데요.
그럼 아래처럼 Optional에 대한 존재 여부를 검사하는 로직을 추가하면 어떨까요?
```suggestion
    private int getMaxSurCharge(final List<Line> boardingLines) {
        OptionalInt max = boardingLines.stream()
                .mapToInt(Line::getSurcharge)
                .max();

        if (max.isPresent()) {
            return max.getAsInt();
        }

        throw new IllegalArgumentException();
    }
```

Reply:
무의식적으로 그냥 Get()으로 넘겨줬네요.,,, 수정하겠습니다!